src/save_data.cpp

```diff
@@ -0,0 +1,92 @@
+#include <ros/ros.h>
+#include <sensor_msgs/Image.h>
+#include <cv_bridge/cv_bridge.h>
+#include <sensor_msgs/image_encodings.h>
+
+#include <opencv2/imgproc/imgproc.hpp>
+#include <opencv2/highgui/highgui.hpp>
+
+#include <message_filters/subscriber.h>
+#include <message_filters/time_synchronizer.h>
+#include <message_filters/sync_policies/approximate_time.h>
+
+#include <moveit/move_group_interface/move_group_interface.h>
+
+#include <iostream>
+#include <string>
+#include <fstream>
+
+using namespace std;
+sensor_msgs::JointState pose_state;
+int initialized = 0;
+
+void Two_image_callback(const sensor_msgs::Image::ConstPtr &image_data1, const sensor_msgs::Image::ConstPtr &image_data2) {
+	if (initialized)
+	{
+		static int count = 0;
+		count++;
+		cout<<"camera count: "<< count << endl;
+		int count_=count;
+
+		sensor_msgs::JointState pose = pose_state;
+
+		cv_bridge::CvImagePtr cv_ptr1;
+		cv_bridge::CvImagePtr cv_ptr2;
+		try
+		{
+			cv_ptr1 = cv_bridge::toCvCopy(image_data1, sensor_msgs::image_encodings::BGR8);
+			cv_ptr2 = cv_bridge::toCvCopy(image_data2, sensor_msgs::image_encodings::BGR8);
+		}
+		catch (cv_bridge::Exception& e)
+		{
+			ROS_ERROR("cv_bridge exception: %s", e.what());
+			return;
+		}
+
+		cv::imwrite( "/home/hand/v4hn/src/hand_demo/data/handpose/" + to_string( count ) + ".jpg", cv_ptr1->image );
+		cv::imwrite( "/home/hand/v4hn/src/hand_demo/data/handpose/" + to_string( ++count ) + ".jpg", cv_ptr2->image );
+
+		ofstream outFile;
+		outFile.open("/home/hand/v4hn/src/hand_demo/data/handpose_data.csv",ios::app);
```

Comment:
Absolute paths do not belong in source code.
Just replace `/home/hand/v4hn/src/hand_demo/data/handpose` by `ros::NodeHandle("~").param("logs", ".")` or something like that.